README.md

```diff
@@ -1 +1,113 @@
-# rna_folding
+# RNA Folding
+
+In biology and chemistry,
+the properties of a molecule are not solely determined by a set of atoms
+but also by the shape of the molecule. 
+In genetics, the shape of an RNA molecule is largely determined by how it bends back on itself. 
+The sequence of A’s, U’s, G’s, and C’s that make up RNA has 
+certain pairs are drawn together to form hydrogen bonds.
+several bonds in a row is called a stem,
+and a stem provides sufficient force to keep the molecule folded together.
+This demo program takes a sequence of RNA and applies a quadratic model to find the optimal stem configuration.
+
+<p align = "center">
+
+![Figure 1!](readme_imgs/Single_Stem.png "This is a title")
+<p align = "center">
+Fig.1 - An RNA sequence with a single stem of length 4.
+</p>
+
+Predicting stems is important to predicting the properties of the RNA molecule.
+However, prediction is made more complicated by the intertwining phenomenon known as psuedoknots.
+The use of a quadratic objective allows for the consideration of pseudoknots, increasing overall accuracy.
+
+<p align = "center">
+
+![Figure 2](readme_imgs/pseudoknot2.png)
+<p align = "center">
+Fig.2 - A pseudoknot formed by a stem of length 3 and a stem of length 5.
+</p>
+
+This demo is loosely based on the work of the paper cited in the References section.
+
+## Usage
+
+To run the demo through a command line interface, type:
+
+```bash
+python RNA_folding.py
+```
+
+The demo prints the optimal stem configuration along with other relavent data. 
+It then saves a plot of the sequence and its bonds as `RNA_plot.png`.
+
+If the 'verbose' parameter is set to its default value of 'True',
+the program will also print information about the model. 
+
+## Problem Formulation
+
+In predicting the stems of an RNA molecule, we build a quadratic model with three contributing factors. 
+
+1. Variables correspond to potential stems, 
+linearly weighted by the negative square of their length, k.
+
+2. Potential pseudoknots correspond to quadratic terms 
+with weight equal to the product of the two lengths 
+times a positive parameter c.
+
+3. Overlapping stems are disallowed, which is enforced by a constraint.
+
+
+![objective](https://latex.codecogs.com/svg.latex?\min_{\bf&space;x}&space;\&space;\sum_i&space;-k_ix_k&space;\&space;&plus;&space;\&space;c\sum_{i,j}&space;k_ik_jx_ix_j) 
```

Comment:
Should this show `k_i^2` in the linear term?

Reply:
Also, is every pair of stems a potential pseudoknot?  The objective function shows coefficients for every pair, but it looks like the code in `pseudoknot_terms` does some filtering.

Reply:
Following up from yesterday's conversation, it sounds like there are still two updates that need to be made to this equation:

1. Linear term should have `k_i^2`
2. Need to express that quadratic terms apply only to stem pairs that are associated with pseudoknots.  E.g., either use a set notation for the sum or use a delta term.